frame/treasury/src/lib.rs

```diff
@@ -371,25 +371,20 @@ impl<T: Trait> Module<T> {
 					return true;
 				}
 
-				// no spent no burn
-				if p.ring_value == RingBalance::<T>::from(0) {
-					miss_any_ring = true;
-				}
-
-				if p.kton_value == KtonBalance::<T>::from(0) {
-					miss_any_kton = true;
-				}
-
```

Comment:
Should infer if balance is zero, because a possible situation is, one of kton or ring is zero.